Source/Blazorise/Utilities/IO/RemoteFileEntryStream.cs

```diff
@@ -52,11 +52,17 @@ public RemoteFileEntryStream( IJSFileModule jsModule, ElementReference elementRe
 
         private async Task FillBuffer( PipeWriter writer, CancellationToken cancellationToken )
         {
-            long offset = 0;
+            long position = 0;
+
+            if ( maxMessageSize < fileEntry.Size )
```

Comment:
ahah variable name is missleading then, should be maxChunkSize just like you described to me. I need to re-test the max size again then, cuz I believe the api was accepting files with gbs.

Reply:
Hmm renaming it to `MaxChunkSize` would be a good idea. It is a breaking change but it makes more sense and is easier to understand. Agree?

Reply:
Also, add a descriptive comment on `MaxMessageSize` / `MaxChunkSize` so that IntelliSense can pick it up.

Reply:
A breaking change? Am I missing smthg, is that exposed? I assumed we had a max file size? Don't we?

Reply:
Yep, I just noticed... I had assumed it was file max size. haha 
But it makes sense to let the developer configure the chunk size, right.

I suggest:

- Adding a new `FileMaxSize` Parameter
- Use the new `MaxLengthExceeded` and check for `FileMaxSize`
- Like you said, Rename `MaxMessageSize` to `MaxChunkSize`, I think that's more clear for developers.
- Rename the enum we just introduced: `FileInvalidReason.UnexpectedBufferLength` to `FileInvalidReason.UnexpectedBufferChunkLength`

Reply:
> Yep, I just noticed... I had assumed it was file max size. haha But it makes sense to let the developer configure the chunk size, right.
> 
> I suggest:
> 
> * Adding a new `FileMaxSize` Parameter
> * (Edit : For backwards compatibility if FileMaxSize is 0, it does not check it)
> * Use the new `MaxLengthExceeded` and check for `FileMaxSize`
> * Like you said, Rename `MaxMessageSize` to `MaxChunkSize`, I think that's more clear for developers.
> * Rename the enum we just introduced: `FileInvalidReason.UnexpectedBufferLength` to `FileInvalidReason.UnexpectedBufferChunkLength`

- Instead of `FileMaxSize` let it be `MaxFileSize`
- Set `MaxFileSize` to `int.MaxValue` for default. Or maybe for safety purposes to 20MB, in bytes?
- This is OK
- Yes, rename it to `MaxChunkSize`. Don't forget to update docs.
- This is also OK.